File: data/compilers_comparison/src/examples/fourth_example/src/func23.c
#include "fourth_example.h" 
array_t* func23(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array25;
   if (pCounter > 0) {
      array25 = vars->data[--pCounter];
      array25->refC++;
      DEBUG_COPY(array25->id);
   } else {
      array25 = (array_t*)malloc(sizeof(array_t));
      array25->size = 403;
      array25->refC = 1;
      array25->id = 25;
      array25->data = (unsigned int*)malloc(array25->size*sizeof(unsigned int));
      memset(array25->data, 0, array25->size*sizeof(unsigned int));
      DEBUG_NEW(array25->id);
   }
   for (int i = 0; i < array25->size; i++) {
      array25->data[i]++;
   }
   for (int i = 0; i < array25->size; i++) {
      if (array25->data[i] == 99) { 
         return array25;
      }
   }
   for (int i = 0; i < array25->size; i++) {
      array25->data[i]--;
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array25;
   array_t* array26 = func31(&params0, rng(), loopsFactor);
   DEBUG_RETURN(array26->id);
   free(params0.data);
   array26->refC--;
   if(array26->refC == 0) {
      free(array26->data);
      free(array26);
      DEBUG_FREE(array26->id);
   }
   return array25;
}

File: data/multilang_comparison/program_nim/src/func23.nim
proc func23*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array124: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array124 = vars.data[pCounter]
      inc array124.refC
      debugCopy(array124.id)
   else:
      array124 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array124.size = 150
      array124.refC = 1
      array124.id = 124
      array124.data = cast[ptr UncheckedArray[uint32]](alloc(array124.size * sizeof(uint32)))
      zeroMem(array124.data, array124.size * sizeof(uint32))
      debugNew(array124.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array124
   var array125: ptr ArrayT = func28(addr params0, getPath(), loopsFactor)
   debugReturn(array125.id)
   dealloc(params0.data)
   let loopLimit38 = (50) div (4) + 1
   for loop38 in 0 ..< loopLimit38:
      discard
      for i in 0 ..< array124.size:
         inc array124.data[i]
   for i in 0 ..< array124.size:
      if array124.data[i] == 22:
         return array124
   dec array124.refC
   if array124.refC == 0:
      dealloc(array124.data)
      dealloc(array124)
      debugFree(array124.id)
   return array125

